src/shared/components/modal/__tests__/ModalProvider.spec.tsx

```diff
@@ -0,0 +1,37 @@
+import * as React from 'react';
+import { Button } from '@patternfly/react-core';
+import { mount } from 'enzyme';
+import { createModalLauncher } from '../createModalLauncher';
+import { useModalLauncher, ModalProvider } from '../ModalProvider';
+
+describe('Modal Provider', () => {
+  let launchMyModal;
+  beforeEach(() => {
+    launchMyModal = createModalLauncher(() => <div>MyModal</div>, {
+      'data-testId': 'my-modal',
+    });
+  });
+  it('should not show modal', () => {
+    const TestComponent = () => {
+      const showModal = useModalLauncher();
+      return <Button onClick={() => showModal(launchMyModal({}))}>Show Modal</Button>;
+    };
+
+    const wrapper = mount(<ModalProvider>{TestComponent}</ModalProvider>);
```

Comment:
it would be better if you could use react-testing-library for the test as we are using that for our tests in hac-dev.